File: src/lessons/recursion/hanoi/TricolorHanoi1Entity.java
package lessons.recursion.hanoi;

import plm.universe.hanoi.HanoiEntity;

public class TricolorHanoi1Entity extends HanoiEntity {
	
	public void run() {
		int src = (Integer)getParam(0);
		int mid = (Integer)getParam(1);
		int dst = (Integer)getParam(2);
		move3(getSlotSize(src)/3, src, mid, dst);
	}

	/* BEGIN TEMPLATE */
	void move3(int height, int src, int mid, int dst) {
		/* BEGIN SOLUTION */
		if (height>0) {
//			System.err.println("move3("+height+","+src+","+dst+")");
			move3(height-1, src, dst, mid);
			move(src,dst);
			move(src,dst);
			move(src,dst);
			move3(height-1, mid, src, dst);
		}
		/* END SOLUTION */
	}

	/* END TEMPLATE */

}

File: resources/exercises/recursion/hanoi/TricolorHanoi1Entity.py
# BEGIN TEMPLATE
def move3(height, src, mid, dst):
	# Your code here
	# BEGIN SOLUTION
	if (height>0):
		move3(height-1, src, dst, mid);
		move(src,dst);
		move(src,dst);
		move(src,dst);
		move3(height-1, mid, src, dst);
	# END SOLUTION
# END TEMPLATE
move3(getSlotSize(getParam(0))/3, getParam(0),getParam(1),getParam(2))